src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -183,6 +183,9 @@ function ($m) {
 
         restore_error_handler();
 
+        // Remove NULL characters to prevent DOMDocument::loadHTML() from stopping the document parsing
+        $content = str_replace("\x00", '', $content);
```

Comment:
The more I think about this, the more I think that there is nothing to fix in Symfony. After all, the crawler just use the underlying PHP libxml binding. That's where this should be fixed instead.